File: backend/app/schemas/analytics.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class InsightRequest(BaseModel):
    query: str = ""
    dataset_id: str | None = None


class InsightResponse(BaseModel):
    summary: str
    model: str


class SampleEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    event_id: str
    user_id: str
    event_type: str
    event_timestamp: datetime
    amount: Decimal
    loaded_at: datetime


class PaginatedEventsResponse(BaseModel):
    items: list[SampleEventResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class EventTypeCount(BaseModel):
    event_type: str
    count: int


class AnalyticsSummaryResponse(BaseModel):
    total_events: int
    total_revenue: Decimal = Field(description="Sum of amount across all events")
    by_event_type: list[EventTypeCount]

File: frontend/src/types/analytics.ts
export interface InsightRequest {
  query?: string;
  dataset_id?: string | null;
}

export interface InsightResponse {
  summary: string;
  model: string;
}

export interface SampleEvent {
  id: number;
  event_id: string;
  user_id: string;
  event_type: string;
  event_timestamp: string;
  amount: string;
  loaded_at: string;
}

export interface PaginatedEventsResponse {
  items: SampleEvent[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface EventTypeCount {
  event_type: string;
  count: number;
}

export interface AnalyticsSummary {
  total_events: number;
  total_revenue: string;
  by_event_type: EventTypeCount[];
}
